### PR TITLE
feat(ui): separate user-defined and framework fields in config tab

### LINKED ISSUE #1144: Separate user-defined fields from base AppConfig and manifest fields in the config tab
## Description

The app config tab shows all fields in a single flat section, mixing the five
framework fields every app inherits from `AppConfig` (`instance_name`,
`log_level`, `app_key`, `forgotten_await_behavior`, `blocking_io_behavior`)
with the fields the app author actually defined. Additionally, two
manifest-level framework fields (`enabled`, `autostart`) are not surfaced in
the config tab at all, even though they're part of how the app is configured.

The fields should be split into two sections: user-defined fields ("App
Settings") shown first, and all framework-managed fields ("Hassette Settings")
shown below — including both the inherited `AppConfig` fields and the manifest
fields `enabled` and `autostart`.

Pydantic's `model_json_schema()` flattens all fields into a single `properties`
dict with no inheritance markers, so the backend must inject this information
explicitly.

## Acceptance Criteria

- [ ] The app config endpoint (`GET /api/apps/{key}/config`) adds a
  `framework_fields` list to the root of the returned schema, containing the
  field names defined on the base `AppConfig` class plus `enabled` and
  `autostart` from `AppManifest`
- [ ] `enabled` and `autostart` values from the manifest are included in the
  config response so they can be rendered in the "Hassette Settings" section
- [ ] The frontend reads `framework_fields` and renders two sections: user
  fields first ("App Settings"), framework fields second ("Hassette Settings")
- [ ] Apps with no user-defined fields (i.e., the class uses `AppConfig`
  directly) show only the "Hassette Settings" section — no empty "App
  Settings" section
- [ ] The "Hassette Settings" section is visually de-emphasised (e.g.
  collapsed by default or rendered at lower visual weight) since these fields
  are rarely the primary concern
- [ ] `config-schema-view.test.tsx` covers the two-section split, the manifest
  fields in the framework section, and the no-user-fields edge case
- [ ] Tests pass

## Affected Areas

- `src/hassette/web/routes/apps.py` — in `_build_app_config_view()`, inject
  `framework_fields: list[str]` by comparing `AppConfig.model_fields.keys()`
  against `app_config_cls.model_fields.keys()`; add `enabled` and `autostart`
  from the manifest to `framework_fields` and include their values in the
  config response
- `src/hassette/web/models.py` — extend `AppConfigResponse` to carry
  `enabled` and `autostart` values for the renderer
- `frontend/src/api/config-view-types.ts` — add `frameworkFields?: string[]`
  to the root schema type
- `frontend/src/components/shared/config-schema-view.tsx` — use
  `frameworkFields` to partition `properties` into two groups; add a
  "Hassette Settings" section label
- `frontend/src/components/shared/config-schema-view.module.css` — style for
  the secondary section
- `frontend/src/components/shared/config-schema-view.test.tsx` — new test
  cases for the partitioned rendering

## Context

`enabled` and `autostart` live on `AppManifest` (`config/classes.py` lines
132–138), not on `AppConfig`. They're TOML-level manifest fields rather than
Pydantic config fields, so they don't appear in the `model_json_schema()` output
and must be injected explicitly alongside the schema. They belong in "Hassette
Settings" because they control framework-level app lifecycle behaviour, not
app-specific logic.

The `framework_fields` list is injected at the endpoint level rather than via
`json_schema_extra` on individual `AppConfig` fields, to avoid coupling the
model to UI rendering concerns.

The global config page (`GET /config`) is not affected — it renders
`HassetteConfig` groups which already have their own section structure.

### PR DESCRIPTION
**Pull request by NodeJSmith**
### Config tab field partitioning

The config tab previously rendered all fields flat — framework-inherited `AppConfig` fields (`instance_name`, `log_level`, `app_key`, `forgotten_await_behavior`, `blocking_io_behavior`) were mixed in with user-defined fields, making it hard to distinguish what a user configured from what the framework manages. Two manifest-level fields (`enabled`, `autostart`) were missing entirely.

The backend now:
- Computes `framework_fields` from `AppConfig.model_fields` + manifest field names, sent in the response so the frontend knows which fields to partition
- Injects `enabled` and `autostart` schema entries into the config schema properties, with their values available from the response envelope
- Returns `autostart` alongside the existing `enabled` field

The frontend now:
- Partitions scalar config fields into "App Settings" (user-defined) and "Hassette Settings" (framework-managed) sections when `frameworkFields` is provided
- De-emphasizes the framework section with reduced opacity and lighter title weight
- Merges manifest values (`enabled`, `autostart`) into the display values before rendering — both the schema-driven and schema-less fallback paths show them
- Preserves the existing "General" section title for the global Config page (no `frameworkFields` → no partitioning)

### Robustness details

- Schema injection uses manifest-fields-first merge order so user-defined fields with the same name win (unlikely but handled)
- Masking continues to scope to `config_props` only — injected boolean manifest fields skip the secret-detection path
- The e2e mock and integration test helper both gained `autostart` support to match the new response shape

Closes #1144

### CODE REVIEW

File: frontend/src/components/shared/config-schema-view.test.tsx [hunk 1]
```diff
@@ -230,6 +230,66 @@ describe("ConfigSchemaView", () => {
...
+    });
+
+    it("applies de-emphasis to the framework section", () => {
+      const { getByTestId } = render(
+        <ConfigSchemaView schema={schema} values={values} frameworkFields={frameworkFields} />,
+      );
+      const fwSection = getByTestId("config-section-hassette-settings");
+      expect(fwSection.className).toMatch(/secondary/);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply frontend/src/components/shared/config-schema-view.test.tsx:311]: Fixed — added the all-framework-managed test case.

File: src/hassette/web/models.py [hunk 2]
```diff
@@ -422,8 +422,10 @@ class AppConfigResponse(BaseModel):
     filename: str
     class_name: str
     enabled: bool
+    autostart: bool = True
     app_config: dict[str, Any] | list[dict[str, Any]]
     config_schema: dict[str, Any] | None = None
+    framework_fields: list[str] = Field(default_factory=list)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/hassette/web/models.py]: Fixed — removed defaults from autostart and framework_fields so the OpenAPI schema marks them required, matching the actual response shape.

File: tests/e2e/mock_fixtures.py [hunk 3]
```diff
@@ -788,6 +788,7 @@ def wire_app_manifest_lookups(hassette, manifests: list[AppManifestInfo]) -> Non
             stub.app_dir.resolve.return_value = real_path.parent
             stub.full_path.exists.return_value = True
         stub.app_config = {"instance_name": f"{m.class_name}.0", "env_prefix": m.app_key + "_"}
+        stub.autostart = getattr(m, "autostart", True)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/e2e/mock_fixtures.py]: Fixed — using m.autostart directly instead of getattr fallback. AppManifestInfo always has this field.
